Frontend/src/pages/registration_edit/components/RegistrationEditor.jsx

```diff
@@ -151,8 +151,23 @@ export default function RegistrationEditor() {
           >
             Update Registration
           </Button>
+          {competitionInfo['using_stripe_payments?'] && (
+            <>
+              <Header>Payment status</Header>
+              {registration.payment_status}
+              {registration.payment_status === 'paid' && (
+                <Button onClick={() => setIsCheckingRefunds(true)}>
+                  {' '}
+                  Show Available Refunds
+                </Button>
+              )}
+            </>
+          )}
+          {isCheckingRefunds && (
+            <Refunds onExit={() => setIsCheckingRefunds(false)} />
+          )}
```

Comment:
Are you doing this rather than `<Refunds isOpen={isCheckingRefunds} onExit={..} />` to avoid the queries in that component from running until open or something like that? I think modals are designed to also be part of the dom, just hidden when closed. One issue I've run into in the past is the open/close animation not working when doing it the way you have it.

Also, what happens if the user closes the modal while the refund is processing? They won't get the 'refund succeeded' message because the component is gone, right? I would consider disabling the ability to close the modal while it's loading.